gitlab.go

```diff
@@ -779,7 +792,6 @@ func (c *Client) NewRequest(method, path string, opt interface{}, options []Opti
 		if fn == nil {
```

Comment:
Is this check still needed?

Reply:
That `if` check needs to stay to make sure we don't cause a panic if someone actively sets `nil` as an `OptionFunc` param value.

See this small example which uses a simplified but similar call tree: https://play.golang.org/p/aT7OXvRt8eT

Reply:
Hm, ok. I'm wondering why this is not considered a bug? If you pass `nil` where a function is expected... normally code just panics when misused.

Reply:
Maybe it has to do with the fact it's a variadic argument. Not sure, but I do know the compiler accepts it so this needs to stay in here.

Reply:
Thinking about it a bit more, I guess the zero value of a function is actually `nil`. So this gives you the same issue: https://play.golang.org/p/PstTInuOrMt